### PR TITLE
Remote Command Implementation Agnostic

### PR DESCRIPTION
**Pull request by malibuworkcrew**
* Removed support for akka remoting
* Updated signatures to allow for custom remoting logic
* Unit tests for both local and 'remote' cases
* Joda time to latest

### CODE REVIEW

File: wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala [hunk 1]
```diff
@@ -41,22 +41,21 @@ object CommandHelper {
...
+    *                    remote command instead of a local one, takes the Input bean and the Command 'id' as parameters
     * @return Result of executing this Command
     */
   def executeCommand[Input <: Product : ClassTag, Output <: Any : ClassTag](id: String,
                                                                            bean: Input,
-                                                                           server:Option[String] = None,
-                                                                           port: Int = 2552,
-                                                                           cm: Option[ActorRef] = None)
+                                                                           cm: Option[ActorRef] = None,
+                                                                           remoteLogic: Option[(String, Input) => Future[Output]] = None)
```
Comment (AlexanderNull) [inline wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala:51]: This signature seems a bit unnecessary. Can you refactor into either two overloaded methods or a separate "executeRemoteCommand" method? There's only really one line of shared code this method currently takes advantage of so having remoteLogic as an option seems like overkill.

File: wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala [hunk 2]
```diff
@@ -164,15 +163,15 @@ trait CommandHelper  { this: Actor =>
...
-   * @param port The port of the remote server defaults to 0, as by default this function deals with local commands
+   * @param bean        the bean that will be passed to the command
+   * @param remoteLogic An additional function that allows one to input logic for executing a
+    *                   remote command instead of a local one, takes the Input bean and the Command 'id' as parameters
    * @return Result of executing this Command
    */
-  def executeCommand[Input<: Product : ClassTag, Output <: Any : ClassTag](id: String, bean: Input, server:Option[String]=None,
-                                                            port:Int=2552)(implicit timeout:Timeout) : Future[Output] = {
+  def executeCommand[Input <: Product : ClassTag, Output <: Any : ClassTag](id: String, bean: Input,
+  remoteLogic: Option[(String, Input) => Future[Output]] = None)(implicit timeout: Timeout): Future[Output] = {
```
Comment (AlexanderNull) [inline wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala:172]: same comment as above

File: wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala (hunk 1 above)
Comment (malibuworkcrew) [reply wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala]: I like it, done!

File: wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala (hunk 2 above)
Comment (malibuworkcrew) [reply wookiee-core/src/main/scala/com/webtrends/harness/command/CommandHelper.scala:183]: Done

**Comment by malibuworkcrew** [comment]
Note that the changes to support SNAPSHOT builds are in this PR now too

**Pull request merged**